### ISSUE TITLE
YAML Resource Enhancement

### ISSUE THREAD

**Issue by rx294** [inspec/inspec#2724]
### Description
Please consider the following enhancement to the YAML resource
I will use the elasticsearch.yml file as an example here
On the yml file for example following three definitions are valid

```
xpack:
  security:
    authc:
      realms.pki1.type: pki
      realms.pki1.order: 1
```
```
xpack.security.authc.realms.pki1.type: pki
xpack.security.authc.realms.pki1.order: 1
```
```
 xpack:
  security:
    authc:
      realms:
        pki1:
          type: pki
          order: 1
```
However the YAML resource exposes these values differently in each case.
```
case 1: yaml('elasticsearch.yml').xpack["security"]["authc"]["realms.pki1.type"]
case 2: yaml('elasticsearch.yml')['xpack.security.authc.realms.pki1.type']
case 3: yaml('elasticsearch.yml').xpack['security']['authc']['realms']['pki1']['type']
```

This will make it very hard to write exhaustive tests to check if the right params are set.

Please consider a enhancement that would allow for tests such as 
yaml('elasticsearch.yml')['xpack.security.authc.realms.pki1.type'] for all possible types of definitions.

### InSpec and Platform Version
2.0.17
### Possible Solutions

May be a solution based on Hashie::Mash can take care of this easily.

@aaronlippold

**Comment by clintoncwolfe** [comment]
Hi @rx294 !   

Yes, I agree that sounds painful, and something should be done to make checks more regular regardless of the YAML source.  I'm not personally familiar with how difficult that would be, so I'll defer on that.

**Comment by burtlo** [comment]
@rx294 It sounds like the elastic search supports this format in YAML but by default I don't think fits the YAML standard. Looking at how Ruby and JavaScript parse these two files and they treat any keys with dot-notation as the whole key. I thought perhaps it was something that was by standard supported in java properties files but I it doesn't appear to be so either.

So it sounds like what you would need here is a custom resource that would further filter the keys. Here is something that I quickly whipped up https://github.com/burtlo/inspec-ohai_resource/blob/master/ohai/libraries/elasticsearch_config.rb that results in supporting the following:

```
describe elasticsearch_config('a.yml') do
  its('xpack.security.authc.realms.pki1.type') { should eq('pki') }
  its('xpack.security.authc.realms.pki1.order') { should eq(1) }
end

describe elasticsearch_config('b.yml') do
  its('xpack.security.authc.realms.pki1.type') { should eq('pki') }
  its('xpack.security.authc.realms.pki1.order') { should eq(1) }
end

describe elasticsearch_config('c.yml') do
  its('xpack.security.authc.realms.pki1.type') { should eq('pki') }
  its('xpack.security.authc.realms.pki1.order') { should eq(1) }
end
```

**Comment by kekaichinose** [comment]
This issue has aged. @rx294 do you have any feedback for @burtlo's workaround? Have you given it a try?

If there is no response in 24 hours, we will go ahead and close the issue.

**Comment by kekaichinose** [comment]
Thank you for tracking down this issue. As with many large open source projects, we have a backlog of bugs and issues of varying user impact. Some issues are perpetually subject to lower priority than others in our backlog. We have reviewed the issue, and we have determined this a very low priority when considered in the context of the entire backlog. We have elected to close it as "Declined" due to competing priorities. We very much appreciate you bringing this to our attention.

Thank you!
__Chef InSpec Team__
